### DISCUSSION TITLE
How do I set env for cargo check/clippy, run configurations, and the terminal

### DISCUSSION THREAD

**Discussion by ShayBox** [intellij-rust/intellij-rust#10521] [Q&A] [answered]
How do I set environment variables globally for cargo check/clippy, run configurations, and the terminal?  

I can set the Terminals environment variables in `Settings > Tools > Terminal > Environment variables > Edit`  
I can edit the Run Configuration environment variables in `Edit Configurations > Cargo > (configuration) > Environment variables > Edit`  

But I see no way to set the environment variables for cargo check/clippy to stop reporting environment variables as not defined at compile time, even though it compiles fine both using Run Configurations and the Terminal.

We don't really need a third place to set them either, can these be pulled from one of the previous locations, or one of the other additional locations such as Path Variables, or the Cargo Run Configuration Template (which doesn't do anything, it doesn't even copy from it when creating new configurations)

I believe pulling environment variables from the currently selected Run Configuration would make the most sense.

**Comment by mchernyavsky** [comment, marked as answer]
Are you talking about cargo check/clippy on-the-fly? If yes, this should help:
<img width="1109" alt="Screenshot 2023-06-01 at 04 58 08" src="https://github.com/intellij-rust/intellij-rust/assets/6079006/888c1e2a-596f-485b-a490-68975bd2c9ab">

**Reply by ShayBox** [reply]
Wow, how did I miss that setting, I go to that page all the time -.-
